ultra/docs/curriculum_learning.md

```diff
@@ -0,0 +1,126 @@
+
+Scenarios can be abstracted over traffic densities, intersection shapes and/or speeds. This means that the levels that contain definitions of scenarios does not need to be very general. We mean that levels can be used to include a definition of **an single scenario**. Here is an example.
+
+- Level easy contains different types of densities and speeds (also applicable to test parameter)
+
+```yaml
+  easy: # (general level)
+    train:
+      # ... irrelevant info is left out
+      intersection_types:
+        2lane_t:
+           percent: 1.0
+           specs: [[50kmh,low-density,0.21],[70kmh,low-density,0.20],[100kmh,low-density,0.20], #61%
+                   [50kmh,mid-density,0.11],[70kmh,mid-density,0.11],[100kmh,mid-density,0.11], #33%,
+                   [50kmh,no-traffic,0.01],[70kmh,no-traffic,0.01],[100kmh,no-traffic,0.01], #3%
+                   [50kmh,high-density,0.01],[70kmh,high-density,0.01],[100kmh,high-density,0.01]] # 3%
+```
+
+- However a level can be very specific (2_lane t intersection with only medium density that flows @ 70kmh). This level will only produce scenarios with that specification
+
+```yaml
+  2t-mid-70: # 2lane_t, mid-density, 70kmh (specific level)
+    train:
+      # ... irrelevant info is left out
+      intersection_types:
+        2lane_t:
+           percent: 1.0
+           specs: [[70kmh,mid-density,1]] # 100%
+```
+
+Due to work in progress with further research into curriculum learning. The dynamic curriculum implementation is separated from the static curriculum. In future updates a more blended architecture will be implemented. For now to use the dynamic curriculum, we must update a few params curriculum config file 
+
+```yaml
+  curriculum:
+    static:
+      # N/A
+    dynamic:
+      toggle: <bool> # Turn on dynamic curriculum settings
+      tasks_levels_used: <list> # All the tasks & levels that "can" be used
+      sampling_rate: <list> # The rate at which the curriculum is updated in terms of episodes
+```
+A specific example with values to describe the dynamic curriculum setup
+
+```yaml
+  curriculum:
+    static:
+      toggle: False
+      # N/A
+    dynamic:
+      toggle: True
+      tasks_levels_used: [["1","no-traffic"], ["1","low-density"], ["1","mid-density"], ["1","high-density"]]
+      sampling_rate: 50
+```
+We can use this method to create these single-scenario levels that only generate a specific range of scenarios. So if the user wants to introduce low-density traffic (with any speed) at a given episode or timestep they can "pull" (build) scenarios from that level and used them. Although, an overlying algorithm or policy will automate the process of building and introducing scenarios to the agents. 
```

Comment:
```suggestion
We can use this method to create these single-scenario levels that only generate a specific range of scenarios. So if the user wants to introduce low-density traffic (with any speed) at a given episode or timestep they can "pull" (build) scenarios from that level and use them. Although, an overlying algorithm or policy will automate the process of building and introducing scenarios to the agents.
```
Also, just curious, what do you mean by "overlying algorithm or policy"? Do you mean something like a teacher agent?